src/osd/OpRequest.cc

```diff
@@ -64,11 +65,25 @@ void OpRequest::_dump(Formatter *f) const
     f->dump_unsigned("tid", m->get_tid());
     f->close_section(); // client_info
   }
+
   {
     f->open_array_section("events");
     std::lock_guard l(lock);
-    for (auto& i : events) {
-      f->dump_object("event", i);
+
+    for (auto i = events.begin(); i != events.end(); ++i) {
+      f->open_object_section("event");
+      f->dump_string("event", i->str);
+      f->dump_stream("time") << i->stamp;
+
+      auto i_next = i;
+      i_next++;
```

Comment:
Right now this can increment past the last event.  You are safe below so long as the last thing you see is "done", but it would be better to be explicit about it.  Maybe change this to something like:

auto i_next = i+1;
if (i_next < events.end()) {
  // Deal with the typical case
} else {
  // Deal with the done case and "not done" case when there are no more events
}

Reply:
Sure! yeah, should have considered corner cases, thanks. Updating.

Reply:
can go with
`      if (i_next < events.end()) {
	f->dump_float("duration", i_next->stamp - i->stamp);
      } else if (i->compare("done") == 0) {
	f->dump_float("duration", i->stamp - get_initiated());
      }
`
this way both conditions will be guarded, and even if i_next, exceeds, nothing would be affected.